allennlp/training/callbacks/callback.py

```diff
@@ -0,0 +1,29 @@
+from typing import TypeVar, Generic
+
+from allennlp.common.registrable import Registrable
+
+TrainerType = TypeVar('TrainerType')  # pylint: disable=invalid-name
+
+class Callback(Registrable, Generic[TrainerType]):
```

Comment:
This class could do with some doccumentation describing the process of implementing a new callback, registering it, what events are exposed within the body of calling it, when they happen etc